packages/common/core-interfaces/src/test/mochaHooks.ts

```diff
@@ -0,0 +1,19 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+/**
+ * This is a minimal hook set to annotate each test case when run for CommonJS.
+ *
+ * @see https://mochajs.org/next/features/root-hook-plugins/
+ *
+ * @remarks Should be kept similar to logic in `@fluid-internal/mocha-test-setup`'s mochaHooks.ts
+ */
+export const mochaHooks = {
+	beforeEach(this: Mocha.Context): void {
+		if (this.currentTest !== undefined && process.env.FLUID_TEST_MODULE_SYSTEM === "CJS") {
```

Comment:
I'm confused by seeing this file here. Since core-interfaces should be nothing but compile-time types, why do we have runtime tests for it? I see they're mostly added by you @jason-ha and at least the ones I skimmed feel weird to me. Can't they be purely type checks that either succeed or fail at build time? I'm almost sure we have things like that in other places (besides the type tests for compatibility testing).